src/Chapter4.hs

```diff
@@ -488,6 +497,27 @@ Implement the 'Applicative' instance for our 'List' type.
   may also need to implement a few useful helper functions for our List
   type.
 -}
+-- helper to join two List together
+join :: List a -> List a -> List a
+join l1 Empty = l1
+join Empty l2 = l2
+join (Cons x xs) l2 = Cons x $ join xs l2
```

Comment:
Usually this function is called `append`, and `join` in Haskell is a function of type `List (List a)` or similar 🙂